numpy/core/tests/test_multiarray.py

```diff
@@ -4728,6 +4728,16 @@ def test_vecself(self):
         p = np.inner(a, a)
         assert_almost_equal(p, 0, decimal=14)
 
+    def test_inner_product_with_various_contiguities(self):
+        # github issue 6532
+        for dt in int, float:
```

Comment:
```
In [1]: np.typecodes
Out[1]: 
{'All': '?bhilqpBHILQPefdgFDGSUVOMm',
 'AllFloat': 'efdgFDG',
 'AllInteger': 'bBhHiIlLqQpP',
 'Character': 'c',
 'Complex': 'FDG',
 'Datetime': 'Mm',
 'Float': 'efdg',
 'Integer': 'bhilqp',
 'UnsignedInteger': 'BHILQP'}
```
So something like
```
for dt in 'bhilqpBHILQPefdgFDG':
```
will get you all the numeric ones. The boolean `?` needs to be checks separatly, as it gives different results.